---
title: "Accessing data from the IGVF Catalog"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Accessing data from the IGVF Catalog}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

# IGVF background

The [Impact of Genomic Variation on Function (IGVF)](https://igvf.org/) 
Consortium,

> aims to understand how genomic variation affects genome function, which in
> turn impacts phenotype. The NHGRI is funding this collaborative program that
> brings together teams of investigators who will use state-of-the-art
> experimental and computational approaches to model, predict, characterize and
> map genome function, how genome function shapes phenotype, and how these
> processes are affected by genomic variation. These joint efforts will produce
> a catalog of the impact of genomic variants on genome function and phenotypes. 

The *IGVF Catalog* described in the last sentence is available through a number
of interfaces, including a [web interface](https://catalog.igvf.org/) as well as 
two programmatic interfaces. In addition, there is a *Data Portal*, where raw
and processed data can be downloaded, with its own
[web interface](https://data.igvf.org/) and 
[programmatic interface](https://github.com/IGVF-DACC/igvf-r-client). 
This package, `rigvf`, focuses on the *Catalog* and not the *Data Portal*.

The *IGVF Catalog* is a form of *knowledge graph*, where the nodes are
biological entities such as variants, genes, pathways, etc. and edges are
relationships between such nodes, e.g. empirically measured effects of variants
on *cis*-regulatory elements (CREs) or on transcripts and proteins. These edges
may have metadata including information about cell type context and information
about how the association was measured, e.g. which experiment or predictive
model.

# This package

This proof-of-concept illustrates how to access the IGVF Catalog.
Only limited functionality is implemented.

# Catalog API

The IGVF offers two programmatic interfaces. The 'catalog'
<https://api.catalog.igvf.org/#> is prefered, with optimized queries
of relevant information. Queries are simple REST requests implemented
using the httr2 package. Here we query variants associated with "GCK";
one could also use, e.g., Ensembl identifiers.

```{r catalog}
rigvf::gene_variants(gene_name = "GCK")

response <- rigvf::gene_variants(gene_id = "ENSG00000106633", verbose = TRUE)
response

response |>
    dplyr::select(`sequence variant`) |>
    tidyr::unnest_wider(`sequence variant`)
```

# ArangoDB API

The 'ArangoDB' REST API provides flexibility but requires greater
understanding of Arango Query Language and the database
schema. Documentation is available in the [database][] under the
'Support' menu item 'REST API' tab using username 'guest' and password
'guestigvfcatalog'.

[database]: https://db.catalog.igvf.org/_db/igvf/_admin/aardvark/index.html#support

The following directly queries the database for variants of an Ensembl
gene id.

```{r example}
rigvf::db_gene_variants("ENSG00000106633", threshold = 0.85)
```

The AQL is

```{r}
aql <- system.file(package = "rigvf", "aql", "gene_variants.aql")
readLines(aql) |> noquote()
```

The help page `?db_queries` outlines other available user-facing
functions. See `?arango` for more developer-oriented information.

```{r}
sessionInfo()
```
