<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Query the 'IGVF' 'ArangoDB' — db_edges • rigvf</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Query the 'IGVF' 'ArangoDB' — db_edges"><meta name="description" content="edges() and nodes() identify edges or nodes in the
data base.
db_gene_variants() locates variants associated with a
(Ensembl) gene identifier."><meta property="og:description" content="edges() and nodes() identify edges or nodes in the
data base.
db_gene_variants() locates variants associated with a
(Ensembl) gene identifier."></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">rigvf</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9003</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/use.html">Accessing data from the IGVF Catalog</a></li>
  </ul></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/mtmorgan/rigvf/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Query the 'IGVF' 'ArangoDB'</h1>
      <small class="dont-index">Source: <a href="https://github.com/mtmorgan/rigvf/blob/main/R/db_queries.R" class="external-link"><code>R/db_queries.R</code></a></small>
      <div class="d-none name"><code>db_queries.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p><code>edges()</code> and <code>nodes()</code> identify edges or nodes in the
data base.</p>
<p><code>db_gene_variants()</code> locates variants associated with a
(Ensembl) gene identifier.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">db_edges</span><span class="op">(</span></span>
<span>  username <span class="op">=</span> <span class="va">rigvf_config</span><span class="op">$</span><span class="fu">get</span><span class="op">(</span><span class="st">"username"</span><span class="op">)</span>,</span>
<span>  password <span class="op">=</span> <span class="va">rigvf_config</span><span class="op">$</span><span class="fu">get</span><span class="op">(</span><span class="st">"password"</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu">db_nodes</span><span class="op">(</span></span>
<span>  username <span class="op">=</span> <span class="va">rigvf_config</span><span class="op">$</span><span class="fu">get</span><span class="op">(</span><span class="st">"username"</span><span class="op">)</span>,</span>
<span>  password <span class="op">=</span> <span class="va">rigvf_config</span><span class="op">$</span><span class="fu">get</span><span class="op">(</span><span class="st">"password"</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu">db_gene_variants</span><span class="op">(</span></span>
<span>  <span class="va">gene_id</span>,</span>
<span>  <span class="va">threshold</span>,</span>
<span>  username <span class="op">=</span> <span class="va">rigvf_config</span><span class="op">$</span><span class="fu">get</span><span class="op">(</span><span class="st">"username"</span><span class="op">)</span>,</span>
<span>  password <span class="op">=</span> <span class="va">rigvf_config</span><span class="op">$</span><span class="fu">get</span><span class="op">(</span><span class="st">"password"</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-username">username<a class="anchor" aria-label="anchor" href="#arg-username"></a></dt>
<dd><p>character(1) ArangoDB user name. Default: "guest".</p></dd>


<dt id="arg-password">password<a class="anchor" aria-label="anchor" href="#arg-password"></a></dt>
<dd><p>character(1) ArangoDB password. Default:
"guestigvfcatalog". A better practice is to use an environment
variable to record the password, rather than encoding in a
script, so <code>password = Sys.getenv("RIGVF_ARANGODB_PASSWORD")</code>.</p></dd>


<dt id="arg-gene-id">gene_id<a class="anchor" aria-label="anchor" href="#arg-gene-id"></a></dt>
<dd><p>character(1) Ensembl gene identifier.</p></dd>


<dt id="arg-threshold">threshold<a class="anchor" aria-label="anchor" href="#arg-threshold"></a></dt>
<dd><p>numeric(1) minimum score associated with the
variant.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p><code>edges()</code> and <code>nodes()</code> return a tibble with the edge or
node name and count of occurrences in the database.</p>
<p><code>db_gene_variants()</code> returns a tibble summarizing variants
associated with the gene.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="fu">db_edges</span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 36 × 2</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    name                                            count</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                                           <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 1</span> variants_variants                          <span style="text-decoration: underline;">5</span>979<span style="text-decoration: underline;">387</span>121</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 2</span> coding_variants_proteins                    239<span style="text-decoration: underline;">108</span>968</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 3</span> variants_coding_variants                    239<span style="text-decoration: underline;">108</span>968</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 4</span> variants_genes_terms                         95<span style="text-decoration: underline;">728</span>528</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 5</span> variants_genes                               95<span style="text-decoration: underline;">728</span>528</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 6</span> regulatory_regions_genes                     30<span style="text-decoration: underline;">197</span>460</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 7</span> regulatory_regions_genes_biosamples_donors   21<span style="text-decoration: underline;">075</span>288</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 8</span> regulatory_regions_genes_biosamples          19<span style="text-decoration: underline;">703</span>741</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 9</span> genes_genes                                   3<span style="text-decoration: underline;">553</span>549</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">10</span> variants_proteins                             2<span style="text-decoration: underline;">927</span>144</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># ℹ 26 more rows</span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu">db_nodes</span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 17 × 2</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    name                       count</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                      <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 1</span> variants              <span style="text-decoration: underline;">1</span>304<span style="text-decoration: underline;">691</span>412</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 2</span> coding_variants        239<span style="text-decoration: underline;">107</span>830</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 3</span> mm_variants            101<span style="text-decoration: underline;">894</span>574</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 4</span> regulatory_regions      17<span style="text-decoration: underline;">138</span>182</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 5</span> mm_regulatory_regions    3<span style="text-decoration: underline;">162</span>620</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 6</span> ontology_terms            <span style="text-decoration: underline;">728</span>868</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 7</span> proteins                  <span style="text-decoration: underline;">294</span>001</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 8</span> transcripts               <span style="text-decoration: underline;">274</span>031</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 9</span> mm_transcripts            <span style="text-decoration: underline;">149</span>547</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">10</span> genes                      <span style="text-decoration: underline;">69</span>222</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">11</span> mm_genes                   <span style="text-decoration: underline;">56</span>941</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">12</span> studies                    <span style="text-decoration: underline;">22</span>690</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">13</span> drugs                       <span style="text-decoration: underline;">4</span>613</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">14</span> pathways                    <span style="text-decoration: underline;">2</span>629</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">15</span> complexes                   <span style="text-decoration: underline;">1</span>527</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">16</span> motifs                       401</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">17</span> donors                       231</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu">db_gene_variants</span><span class="op">(</span><span class="st">"ENSG00000106633"</span>, <span class="fl">0.85</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 40 × 9</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    `_key`              `_id` `_from` `_to` `_rev` `score:long` source source_url</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>               <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 1</span> genic_chr7_4415452… regu… regula… gene… _g5CU…        0.989 ENCOD… https://w…</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 2</span> promoter_chr7_4415… regu… regula… gene… _g5CU…        0.869 ENCOD… https://w…</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 3</span> genic_chr7_4414584… regu… regula… gene… _g5CU…        0.948 ENCOD… https://w…</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 4</span> promoter_chr7_4415… regu… regula… gene… _g5CU…        1.00  ENCOD… https://w…</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 5</span> promoter_chr7_4415… regu… regula… gene… _g5CU…        1.00  ENCOD… https://w…</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 6</span> intergenic_chr7_44… regu… regula… gene… _g5CU…        0.959 ENCOD… https://w…</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 7</span> genic_chr7_4415544… regu… regula… gene… _g5CV…        0.942 ENCOD… https://w…</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 8</span> promoter_chr7_4415… regu… regula… gene… _g5CV…        0.929 ENCOD… https://w…</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 9</span> intergenic_chr7_44… regu… regula… gene… _g5CV…        0.936 ENCOD… https://w…</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">10</span> promoter_chr7_4415… regu… regula… gene… _g5CW…        0.966 ENCOD… https://w…</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># ℹ 30 more rows</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># ℹ 1 more variable: biological_context &lt;chr&gt;</span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Martin Morgan, Michael Love.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer></div>





  </body></html>

