<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Accessing data from the IGVF Catalog • rigvf</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Accessing data from the IGVF Catalog">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">rigvf</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9003</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/use.html">Accessing data from the IGVF Catalog</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/mtmorgan/rigvf/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Accessing data from the IGVF Catalog</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/mtmorgan/rigvf/blob/main/vignettes/use.Rmd" class="external-link"><code>vignettes/use.Rmd</code></a></small>
      <div class="d-none name"><code>use.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="igvf-background">IGVF background<a class="anchor" aria-label="anchor" href="#igvf-background"></a>
</h2>
<p>The <a href="https://igvf.org/" class="external-link">Impact of Genomic Variation on
Function (IGVF)</a> Consortium,</p>
<blockquote>
<p>aims to understand how genomic variation affects genome function,
which in turn impacts phenotype. The NHGRI is funding this collaborative
program that brings together teams of investigators who will use
state-of-the-art experimental and computational approaches to model,
predict, characterize and map genome function, how genome function
shapes phenotype, and how these processes are affected by genomic
variation. These joint efforts will produce a catalog of the impact of
genomic variants on genome function and phenotypes.</p>
</blockquote>
<p>The <em>IGVF Catalog</em> described in the last sentence is available
through a number of interfaces, including a <a href="https://catalog.igvf.org/" class="external-link">web interface</a> as well as two
programmatic interfaces. In addition, there is a <em>Data Portal</em>,
where raw and processed data can be downloaded, with its own <a href="https://data.igvf.org/" class="external-link">web interface</a> and <a href="https://github.com/IGVF-DACC/igvf-r-client" class="external-link">programmatic
interface</a>. This package, <code>rigvf</code>, focuses on the
<em>Catalog</em> and not the <em>Data Portal</em>.</p>
<p>The <em>IGVF Catalog</em> is a form of <em>knowledge graph</em>,
where the nodes are biological entities such as variants, genes,
pathways, etc. and edges are relationships between such nodes,
e.g. empirically measured effects of variants on <em>cis</em>-regulatory
elements (CREs) or on transcripts and proteins. These edges may have
metadata including information about cell type context and information
about how the association was measured, e.g. which experiment or
predictive model.</p>
</div>
<div class="section level2">
<h2 id="this-package">This package<a class="anchor" aria-label="anchor" href="#this-package"></a>
</h2>
<p>This proof-of-concept illustrates how to access the IGVF Catalog.
Only limited functionality is implemented.</p>
</div>
<div class="section level2">
<h2 id="catalog-api">Catalog API<a class="anchor" aria-label="anchor" href="#catalog-api"></a>
</h2>
<p>The IGVF offers two programmatic interfaces. The ‘catalog’ <a href="https://api.catalog.igvf.org/#" class="external-link uri">https://api.catalog.igvf.org/#</a> is prefered, with
optimized queries of relevant information. Queries are simple REST
requests implemented using the httr2 package. Here we query variants
associated with “GCK”; one could also use, e.g., Ensembl
identifiers.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">rigvf</span><span class="fu">::</span><span class="fu"><a href="../reference/catalog.html">gene_variants</a></span><span class="op">(</span>gene_name <span class="op">=</span> <span class="st">"GCK"</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 25 × 9</span></span></span>
<span><span class="co">#&gt;    `sequence variant`      gene  label log10pvalue effect_size source source_url</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> variants/8c6a683829bcb… gene… eQTL         4.89       0.274 GTEx   https://s…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> variants/cf796b5a16212… gene… eQTL         5.76       0.221 GTEx   https://s…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> variants/9a36af4633321… gene… eQTL         6.17      -<span style="color: #BB0000;">0.266</span> GTEx   https://s…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> variants/2fefe07a0750b… gene… eQTL         3.69       0.158 GTEx   https://s…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> variants/ab6df1152a643… gene… eQTL        16.9       -<span style="color: #BB0000;">0.353</span> GTEx   https://s…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> variants/92833b52621e5… gene… eQTL         4.86      -<span style="color: #BB0000;">0.170</span> GTEx   https://s…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> variants/bceca4e6ac3cd… gene… eQTL         4.63      -<span style="color: #BB0000;">0.340</span> GTEx   https://s…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> variants/0a8ba63e5451a… gene… eQTL         4.94       0.215 GTEx   https://s…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> variants/80f639e0da643… gene… eQTL         6.59      -<span style="color: #BB0000;">0.330</span> GTEx   https://s…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> variants/7f4ca6f1cfd70… gene… eQTL         4.10      -<span style="color: #BB0000;">0.165</span> GTEx   https://s…</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 15 more rows</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 2 more variables: biological_context &lt;chr&gt;, chr &lt;chr&gt;</span></span></span>
<span></span>
<span><span class="va">response</span> <span class="op">&lt;-</span> <span class="fu">rigvf</span><span class="fu">::</span><span class="fu"><a href="../reference/catalog.html">gene_variants</a></span><span class="op">(</span>gene_id <span class="op">=</span> <span class="st">"ENSG00000106633"</span>, verbose <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">response</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 25 × 9</span></span></span>
<span><span class="co">#&gt;    `sequence variant` gene              label log10pvalue effect_size source</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;list&gt;</span>             <span style="color: #949494; font-style: italic;">&lt;list&gt;</span>            <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> <span style="color: #949494;">&lt;named list [14]&gt;</span>  <span style="color: #949494;">&lt;named list [11]&gt;</span> eQTL         4.89       0.274 GTEx  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> <span style="color: #949494;">&lt;named list [14]&gt;</span>  <span style="color: #949494;">&lt;named list [11]&gt;</span> eQTL         5.76       0.221 GTEx  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> <span style="color: #949494;">&lt;named list [14]&gt;</span>  <span style="color: #949494;">&lt;named list [11]&gt;</span> eQTL         6.17      -<span style="color: #BB0000;">0.266</span> GTEx  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> <span style="color: #949494;">&lt;named list [14]&gt;</span>  <span style="color: #949494;">&lt;named list [11]&gt;</span> eQTL         3.69       0.158 GTEx  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> <span style="color: #949494;">&lt;named list [14]&gt;</span>  <span style="color: #949494;">&lt;named list [11]&gt;</span> eQTL        16.9       -<span style="color: #BB0000;">0.353</span> GTEx  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> <span style="color: #949494;">&lt;named list [14]&gt;</span>  <span style="color: #949494;">&lt;named list [11]&gt;</span> eQTL         4.86      -<span style="color: #BB0000;">0.170</span> GTEx  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> <span style="color: #949494;">&lt;named list [14]&gt;</span>  <span style="color: #949494;">&lt;named list [11]&gt;</span> eQTL         4.63      -<span style="color: #BB0000;">0.340</span> GTEx  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> <span style="color: #949494;">&lt;named list [14]&gt;</span>  <span style="color: #949494;">&lt;named list [11]&gt;</span> eQTL         4.94       0.215 GTEx  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> <span style="color: #949494;">&lt;named list [14]&gt;</span>  <span style="color: #949494;">&lt;named list [11]&gt;</span> eQTL         6.59      -<span style="color: #BB0000;">0.330</span> GTEx  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> <span style="color: #949494;">&lt;named list [14]&gt;</span>  <span style="color: #949494;">&lt;named list [11]&gt;</span> eQTL         4.10      -<span style="color: #BB0000;">0.165</span> GTEx  </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 15 more rows</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 3 more variables: source_url &lt;chr&gt;, biological_context &lt;chr&gt;, chr &lt;chr&gt;</span></span></span>
<span></span>
<span><span class="va">response</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">`sequence variant`</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu">tidyr</span><span class="fu">::</span><span class="fu"><a href="https://tidyr.tidyverse.org/reference/unnest_wider.html" class="external-link">unnest_wider</a></span><span class="op">(</span><span class="va">`sequence variant`</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 25 × 14</span></span></span>
<span><span class="co">#&gt;    organism     `_id`    chr      pos rsid  ref   alt   spdi  hgvs  qual  filter</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span> </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> Homo sapiens 8c6a683… chr7  4.41<span style="color: #949494;">e</span>7 rs25… G     A     NC_0… NC_0… .     <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> Homo sapiens cf796b5… chr7  4.41<span style="color: #949494;">e</span>7 rs29… T     C     NC_0… NC_0… .     <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> Homo sapiens 9a36af4… chr7  4.41<span style="color: #949494;">e</span>7 rs11… GT    G     NC_0… NC_0… .     <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> Homo sapiens 2fefe07… chr7  4.43<span style="color: #949494;">e</span>7 rs28… G     A     NC_0… NC_0… .     <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> Homo sapiens ab6df11… chr7  4.41<span style="color: #949494;">e</span>7 rs22… C     G     NC_0… NC_0… .     <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> Homo sapiens 92833b5… chr7  4.41<span style="color: #949494;">e</span>7 rs41… G     A     NC_0… NC_0… .     <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> Homo sapiens bceca4e… chr7  4.40<span style="color: #949494;">e</span>7 rs76… T     G     NC_0… NC_0… .     <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> Homo sapiens 0a8ba63… chr7  4.41<span style="color: #949494;">e</span>7 rs25… A     G     NC_0… NC_0… .     <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> Homo sapiens 80f639e… chr7  4.41<span style="color: #949494;">e</span>7 rs14… A     AG    NC_0… NC_0… .     <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> Homo sapiens 7f4ca6f… chr7  4.42<span style="color: #949494;">e</span>7 rs29… A     T     NC_0… NC_0… .     <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 15 more rows</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 3 more variables: annotations &lt;list&gt;, source &lt;chr&gt;, source_url &lt;chr&gt;</span></span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="arangodb-api">ArangoDB API<a class="anchor" aria-label="anchor" href="#arangodb-api"></a>
</h2>
<p>The ‘ArangoDB’ REST API provides flexibility but requires greater
understanding of Arango Query Language and the database schema.
Documentation is available in the <a href="https://db.catalog.igvf.org/_db/igvf/_admin/aardvark/index.html#support" class="external-link">database</a>
under the ‘Support’ menu item ‘REST API’ tab using username ‘guest’ and
password ‘guestigvfcatalog’.</p>
<p>The following directly queries the database for variants of an
Ensembl gene id.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">rigvf</span><span class="fu">::</span><span class="fu"><a href="../reference/db_queries.html">db_gene_variants</a></span><span class="op">(</span><span class="st">"ENSG00000106633"</span>, threshold <span class="op">=</span> <span class="fl">0.85</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 40 × 9</span></span></span>
<span><span class="co">#&gt;    `_key`              `_id` `_from` `_to` `_rev` `score:long` source source_url</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>               <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> genic_chr7_4415452… regu… regula… gene… _g5CU…        0.989 ENCOD… https://w…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> promoter_chr7_4415… regu… regula… gene… _g5CU…        0.869 ENCOD… https://w…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> genic_chr7_4414584… regu… regula… gene… _g5CU…        0.948 ENCOD… https://w…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> promoter_chr7_4415… regu… regula… gene… _g5CU…        1.00  ENCOD… https://w…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> promoter_chr7_4415… regu… regula… gene… _g5CU…        1.00  ENCOD… https://w…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> intergenic_chr7_44… regu… regula… gene… _g5CU…        0.959 ENCOD… https://w…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> genic_chr7_4415544… regu… regula… gene… _g5CV…        0.942 ENCOD… https://w…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> promoter_chr7_4415… regu… regula… gene… _g5CV…        0.929 ENCOD… https://w…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> intergenic_chr7_44… regu… regula… gene… _g5CV…        0.936 ENCOD… https://w…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> promoter_chr7_4415… regu… regula… gene… _g5CW…        0.966 ENCOD… https://w…</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 30 more rows</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 1 more variable: biological_context &lt;chr&gt;</span></span></span></code></pre></div>
<p>The AQL is</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">aql</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span>package <span class="op">=</span> <span class="st">"rigvf"</span>, <span class="st">"aql"</span>, <span class="st">"gene_variants.aql"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/readLines.html" class="external-link">readLines</a></span><span class="op">(</span><span class="va">aql</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/noquote.html" class="external-link">noquote</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] FOR l IN regulatory_regions_genes     </span></span>
<span><span class="co">#&gt; [2]     FILTER l._to == @geneid           </span></span>
<span><span class="co">#&gt; [3]     FILTER l.`score:long` &gt; @threshold</span></span>
<span><span class="co">#&gt; [4]     RETURN l</span></span></code></pre></div>
<p>The help page <code>?db_queries</code> outlines other available
user-facing functions. See <code><a href="../reference/arango.html">?arango</a></code> for more
developer-oriented information.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.4.1 (2024-06-14)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 22.04.5 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span></span>
<span><span class="co">#&gt;  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span></span>
<span><span class="co">#&gt; [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: UTC</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;  [1] jsonlite_1.8.9    dplyr_1.1.4       compiler_4.4.1    tidyselect_1.2.1 </span></span>
<span><span class="co">#&gt;  [5] rigvf_0.0.0.9003  tidyr_1.3.1       jquerylib_0.1.4   systemfonts_1.1.0</span></span>
<span><span class="co">#&gt;  [9] textshaping_0.4.0 yaml_2.3.10       fastmap_1.2.0     R6_2.5.1         </span></span>
<span><span class="co">#&gt; [13] rjsoncons_1.3.1   generics_0.1.3    curl_5.2.3        httr2_1.0.4      </span></span>
<span><span class="co">#&gt; [17] knitr_1.48        tibble_3.2.1      desc_1.4.3        bslib_0.8.0      </span></span>
<span><span class="co">#&gt; [21] pillar_1.9.0      rlang_1.1.4       utf8_1.2.4        cachem_1.1.0     </span></span>
<span><span class="co">#&gt; [25] xfun_0.47         fs_1.6.4          sass_0.4.9        memoise_2.0.1    </span></span>
<span><span class="co">#&gt; [29] cli_3.6.3         withr_3.0.1       pkgdown_2.1.1     magrittr_2.0.3   </span></span>
<span><span class="co">#&gt; [33] digest_0.6.37     rappdirs_0.3.3    lifecycle_1.0.4   vctrs_0.6.5      </span></span>
<span><span class="co">#&gt; [37] evaluate_1.0.0    glue_1.7.0        whisker_0.4.1     ragg_1.3.3       </span></span>
<span><span class="co">#&gt; [41] fansi_1.0.6       rmarkdown_2.28    purrr_1.0.2       tools_4.4.1      </span></span>
<span><span class="co">#&gt; [45] pkgconfig_2.0.3   htmltools_0.5.8.1</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Martin Morgan, Michael Love.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
